template/.github/workflows/nightly_at_main.yml

```diff
@@ -20,6 +20,10 @@ jobs:
   tests:
     name: Tests
     needs: setup
+    env:
+        # Security note! The secrets are only added to workflows that run on trusted branches (main). If secrets are accessible in workflows that run on untrusted branches they can be extracted, see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#exfiltrating-data-from-a-runner.
+        ESS_PROTECTED_FILESTORE_USERNAME: ${{ secrets.ESS_PROTECTED_FILESTORE_USERNAME }}
+        ESS_PROTECTED_FILESTORE_PASSWORD: ${{ secrets.ESS_PROTECTED_FILESTORE_PASSWORD }}
```

Comment:
We were recently trying to remove Scipp and ESS-specifics from the copier-template. This seems to counter that effort?

Reply:
Yes it does 🤔 how do we fix that?

Reply:
You could move it into the ess template. But then it could not be used in ScippNeutron and ScippNexus.

Reply:
Yes. The templates will be merged into the same repo. GH CI won't care where the files ultimately come from.

Reply:
But then I have to move the entire workflow to `ess_template`. Or can I just move the part that concerns the environment variables?

Reply:
Is there a way to move the env section to a custom file that gets included in the workflow, and write the custom file content only if the template is rendered for our projects?

Reply:
Maybe, is there an example of a similar mechanism somewhere else in the copier template?